### PR TITLE
Declutter docstrings fmpz and change `fmpz_divisible_si` and `fmpz_mod_ui`

### PR DESCRIPTION
**Pull request by albinahlback**
Modify `fmpz_divisible_si` and its docstring in order to work regardless of the slongs sign.

Also inline `fmpz_mod_ui` and remove its argument `f`, since it basically just calls a `fdiv` function. Here I clarified its and `fmpz_mod`'s docstring.

Note that this will break Flint. I think it's a good idea to introduce this change, and I could change the rest of the repository to this, but please let me know what to do.

### CODE REVIEW

**Comment by wbhart** [comment]
You can't change the definition of a basic Flint function that's been there for years. It won't just break Flint but all the code using Flint!

File: doc/source/fmpz.rst [hunk 1]
```diff
@@ -596,14 +592,8 @@ Comparison
...
-    otherwise returns `0`.
-
 .. function:: int fmpz_cmp_ui(const fmpz_t f, ulong g)
 
-    Returns a negative value if `f < g`, positive value if `g < f`, 
-    otherwise returns `0`.
-
 .. function:: int fmpz_cmp_si(const fmpz_t f, slong g)
 
     Returns a negative value if `f < g`, positive value if `g < f`,
```
Comment (wbhart) [inline doc/source/fmpz.rst:599]: The other docstring is the better one as it says what happens otherwise.

File: fmpz/divisible_si.c [hunk 2]
```diff
@@ -14,22 +14,23 @@
 #include "ulong_extras.h"
 #include "fmpz.h"
 
-int fmpz_divisible_si(const fmpz_t x, slong p)
+int
+fmpz_divisible_si(const fmpz_t f, slong g)
```
Comment (wbhart) [inline fmpz/divisible_si.c:18]: This will need new test code which exercises it for the new inputs that are supported.

**Review by wbhart — CHANGES_REQUESTED** [review]

File: doc/source/fmpz.rst (hunk 1 above)
Comment (albinahlback) [reply doc/source/fmpz.rst:599]: I haven't changed the docstring for this function, and I don't see a difference between this docstring and the other ones for `fmpz_cmp` and `fmpz_cmp_ui`. Or did you mean some other docstring perhaps?

Comment (wbhart) [reply doc/source/fmpz.rst:599]: Oh I see the diff cuts off there. My mistake.

File: fmpz/divisible_si.c (hunk 2 above)
Comment (albinahlback) [reply fmpz/divisible_si.c:19]: Done.

**Comment by albinahlback** [comment]
I now reverted `fmpz_mod_ui` back to the previous state and only inlined it.

**Comment by albinahlback** [comment]
Forgot to call FLINT_ABS before calling GMP's divisibility test in the new test. This PR should be final now unless you have any other comments.

**Comment by wbhart** [comment]
Looks ok now. Thanks. I'll merge tomorrow if no objections.

**Comment by wbhart** [comment]
Although note that the tests are now failing.

**Comment by albinahlback** [comment]
Sorry, missed changing a `fmpz_mul_ui` to `fmpz_mul_si` in the tests. They now pass on my computer.

**Comment by wbhart** [comment]
Looks good now, thanks.

**Pull request merged**